### ISSUE TITLE
Dialog: Performing "onMouseDown" inside the Dialog and "onMouseUp" outside of it closes the dialog when "dismissableMask" is true

### ISSUE THREAD

**Issue by VentuzVictor** [primefaces/primereact#3962]
### Describe the bug

Having "dismissableMask" as "true" in a Dialog, it closes automatically when an "onMouseDown" is performed inside the Dialog, for example selecting text inside an InputNumber / p and then releasing the mouse ("onMouseUp") outside the Dialog.

This causes bad user interaction and assuming that the Dialog contains a Form with several components, editing the InputNumber quickly and selecting the text inside causes it to close unexpectedly (see gif).

![dialog_err](https://user-images.githubusercontent.com/71069615/212700633-b2ed1327-65d8-4982-873a-6f49a61c8373.gif)

### Reproducer

https://codesandbox.io/s/primereact-test-forked-uym2ut?file=/src/index.js:750-765

### PrimeReact version

8.7.3

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Browser(s)

Chrome 108

### Steps to reproduce the behavior

1. Open the dialog in the forked project.
2. Write something in the input number
3. Select the input text quickly and release the mouse outside the dialog.

-- It closes unexpectadly --.

### Expected behavior

The dialog should remain displayed since officially an "onClick" was not performed outside of it ("onMouseDown" was performed inside and "onMouseUp" was performed outside).

**Comment by melloware** [comment]
Thanks for the reproducer its interesting I can make it happen with Create React App but not with Next.  I have a fix that will work for both.

**Comment by VentuzVictor** [comment]
Is there a possibility to release this fix in a minor version of 9.x.x instead of waiting until 10? Thanks

**Comment by melloware** [comment]
its OK I have assigned PrimeTek to review my PR.

**Comment by VentuzVictor** [comment]
Hi @melloware , just to clarify, are we waiting for their response on the pull request you made in January, or did you re-request it after my question last week?

**Comment by melloware** [comment]
Yep PrimeTek was working on major changes like today's 9.3.0 release allowing custom icons everywhere.  So they have not been merging PR's

**Comment by VentuzVictor** [comment]
But I have a feeling that they are only going to look at PR's that correspond to milestone 9.x.x, this issue is tagged with Milestone 10.0.0, hence my original question.

> Is there a possibility to release this fix in a minor version of 9.x.x instead of waiting until 10? Thanks
> 
> (Sorry, I inadvertently closed and reopened the issue)

**Comment by melloware** [comment]
@VentuzVictor if you need this fixed sooner I suggest you investigate PRO support: https://primereact.org/support/

**Comment by VentuzVictorPerez** [comment]
Cool, thanks!
